tests/phpunit/CRM/Core/BAO/MessageTemplateTest.php

```diff
@@ -164,7 +164,14 @@ public function testContactTokens(): void {
     $expected .= $this->getExpectedContactOutput($address['id'], $tokenData, $messageContent['html']);
     $this->assertEquals($expected, $messageContent['html']);
     $this->assertEquals($expected, $messageContent['text']);
-    $this->assertEquals(rtrim(str_replace("\n", ' ', $expected)), $messageContent['subject']);
+    $checksum_position = strpos($messageContent['subject'], 'cs=');
```

Comment:
It might be good to validate `$this->assertTrue($checksum_position !== FALSE)`, since the substr's will happily accept FALSE, and the explodes lower down will end up working on the whole string which would give a false positive.

Reply:
Ok updated now @demeritcowboy